plugin/src/main/java/org/opennms/velocloud/events/VelocloudEventIngestor.java

```diff
@@ -0,0 +1,348 @@
+        builder.addParameter(ImmutableEventParameter.newInstance("category", enterpriseEvent.getCategory()));
+        builder.addParameter(ImmutableEventParameter.newInstance("enterpriseUsername", enterpriseEvent.getEnterpriseUsername()));
+        builder.addParameter(ImmutableEventParameter.newInstance("detail", enterpriseEvent.getDetail()));
+        builder.addParameter(ImmutableEventParameter.newInstance("id", String.valueOf(enterpriseEvent.getId())));
+
+        if (enterpriseEvent.getEventTime() != null) {
+            builder.setTime(Date.from(enterpriseEvent.getEventTime().toInstant()));
+            builder.addParameter(ImmutableEventParameter.newInstance("eventTime", DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(enterpriseEvent.getEventTime())));
+        } else {
+            builder.addParameter(ImmutableEventParameter.newInstance("eventTime", null));
+        }
+
+        eventForwarder.sendSync(builder.build());
+    }
+
+    private void processCustomerEvents(final Instant start, final Instant end, final String foreignSource, final VelocloudApiCustomerClient velocloudApiCustomerClient) throws VelocloudApiException {
+        final List<EnterpriseEvent> enterpriseEventList = velocloudApiCustomerClient.getEnterpriseEvents(start, end);
+        LOG.debug(enterpriseEventList.size() + " enterprise events found.");
+
+        for (final EnterpriseEvent enterpriseEvent : enterpriseEventList) {
+            if (!Strings.isNullOrEmpty(enterpriseEvent.getEdgeName())) {
+                processEvent(foreignSource, enterpriseEvent.getEdgeName(), enterpriseEvent);
+            } else {
+                LOG.debug("ignoring enterprise event #" + enterpriseEvent.getId() + " due to missing edgeName.");
+            }
+        }
+    }
+
+    private void processPartnerEvents(final Instant start, final Instant end, final String foreignSource, final VelocloudApiPartnerClient velocloudApiPartnerClient) throws VelocloudApiException {
+        final List<ProxyEvent> enterpriseEventList = velocloudApiPartnerClient.getProxyEvents(start, end);
+        LOG.debug(enterpriseEventList.size() + " proxy events found.");
+        for (final ProxyEvent proxyEvent : enterpriseEventList) {
+            if (!Strings.isNullOrEmpty(proxyEvent.getGatewayName())) {
+                processEvent(foreignSource, proxyEvent.getGatewayName(), proxyEvent);
+            } else {
+                LOG.debug("ignoring proxy event #" + proxyEvent.getId() + " due to missing gatewayName.");
+            }
+        }
+    }
+
+    private String getMetadataForKey(final String key, final List<MetaData> metaDataList) {
+        for (final MetaData metaData : metaDataList) {
+            if (AbstractRequisitionProvider.VELOCLOUD_METADATA_CONTEXT.equals(metaData.getContext()) && key.equals(metaData.getKey())) {
+                return metaData.getValue();
+            }
+        }
+        return null;
+    }
+
+    private String parseJson(String detail, String key) {
```

Comment:
Please add a comment why this is as ugly as this.

Reply:
Replaced with Yaml parser